packages/fuels-accounts/src/provider.rs

```diff
@@ -491,19 +491,27 @@ impl Provider {
         &self,
         contract_id: &Bech32ContractId,
     ) -> Result<HashMap<AssetId, u64>> {
-        // We don't paginate results because there are likely at most ~100 different assets in one
-        // wallet
-        let pagination = PaginationRequest {
+        let mut pagination = PaginationRequest {
             cursor: None,
-            results: 9999,
+            results: 512,
```

Comment:
Is this something we can query from the node? If possible maybe upon `provider.connect` and cache it.

Reply:
It is just arbitrary for now.